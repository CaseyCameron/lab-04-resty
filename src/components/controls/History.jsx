import React from 'react';
import PropTypes from 'prop-types';

const History = ({ method, urlValue }) => {
  return (
    <>
      <p>{method}</p>
      <p>{urlValue}</p>
    </>
  )
};

History.propTypes = {
  method: PropTypes.string.isRequired,
  urlValue: PropTypes.string.isRequired,
};

export default History;