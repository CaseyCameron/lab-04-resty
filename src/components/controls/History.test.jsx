import React from 'react';
import { render, screen } from '@testing-library/react';
import History from './History';

describe('Method History component', () => {
  it('Displays a history element', () => {
    const { asFragment } = render(
      <History
        method='GET'
        urlValue='https://last-airbender-api.herokuapp.com/api/v1/characters/5cf5679a915ecad153ab68c8'
      />);

    expect(asFragment()).toMatchSnapshot();
  });
});