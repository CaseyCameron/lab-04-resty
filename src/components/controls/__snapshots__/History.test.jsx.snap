// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Method History component Displays a history element 1`] = `
<DocumentFragment>
  <p>
    GET
  </p>
  <p>
    https://last-airbender-api.herokuapp.com/api/v1/characters/5cf5679a915ecad153ab68c8
  </p>
</DocumentFragment>
`;
